arangod/RocksDBEngine/RocksDBV8Functions.cpp

```diff
@@ -221,6 +223,89 @@ static void JS_CollectionRevisionTreeCorrupt(v8::FunctionCallbackInfo<v8::Value>
   TRI_V8_RETURN_UNDEFINED();
   TRI_V8_TRY_CATCH_END
 }
+
+static void JS_CollectionRevisionTreeVerification(v8::FunctionCallbackInfo<v8::Value> const& args) {
+  TRI_V8_TRY_CATCH_BEGIN(isolate);
+  v8::HandleScope scope(isolate);
+
+  auto* collection = UnwrapCollection(isolate, args.Holder());
+
+  if (!collection) {
+    TRI_V8_THROW_EXCEPTION_INTERNAL("cannot extract collection");
+  }
+
+  std::unique_ptr<containers::RevisionTree> storedTree;
+  std::unique_ptr<containers::RevisionTree> computedTree;
+
+  {
+    TRI_vocbase_t& vocbase = collection->vocbase();
+    auto& server = vocbase.server();
+    RocksDBEngine& engine = server.getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+    RocksDBReplicationManager* manager = engine.replicationManager();
+    double ttl = 3600;
+    RocksDBReplicationContext* ctx 
+      = manager->createContext(engine, ttl, SyncerId{17}, ServerId{17}, "");
+    if (ctx == nullptr) {
+      TRI_V8_THROW_EXCEPTION_INTERNAL("Could not create RocksDBReplicationContext");
+    }
+    RocksDBReplicationContextGuard guard(manager, ctx);
+    auto* physical = toRocksDBCollection(*collection);
+    auto batchId = ctx->id();
+    storedTree = physical->revisionTree(batchId);
+    computedTree = physical->computeRevisionTree(batchId);
+    ctx->setDeleted();
+  }
+
+  VPackBuilder builder;
+  { 
+    VPackObjectBuilder guard(&builder);
+    if (storedTree != nullptr) {
+      builder.add(VPackValue("storedTree"));
+      storedTree->serialize(builder);
+    } else {
+      builder.add("storedTree", VPackValue(false));
+    }
+    if (computedTree != nullptr) {
+      builder.add(VPackValue("computedTree"));
+      computedTree->serialize(builder);
+    } else {
+      builder.add("computedTree", VPackValue(false));
+    }
+    if (storedTree != nullptr && computedTree != nullptr) {
+      try {
+        std::vector<std::pair<uint64_t, uint64_t>> diff
+          = computedTree->diff(*storedTree);
+        builder.add("equal", VPackValue(diff.empty()));
+      } catch(std::exception const& exc) {
+        builder.add("equal", VPackValue(std::string(exc.what())));
```

Comment:
We store the exception message in the attribute `equal`?

Reply:
I am now storing it in `error` instead.